hybrasyl/Objects/User.cs

```diff
@@ -2263,25 +2263,222 @@ public override void Attack(Castable castObject)
+                                        Map.EntityTree.GetObjects(new Rectangle(this.X - intent.Radius, this.Y, intent.Radius, 1))
+                                            .Where(
+                                                obj =>
+                                                    obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                                }
+                                    break;
+                                case Direction.East:
+                                {
+                                    //facing east, attack south
+                                    possibleTargets.AddRange(
+                                        Map.EntityTree.GetObjects(new Rectangle(this.X, this.Y - intent.Radius, 1, intent.Radius))
+                                            .Where(
+                                                obj =>
+                                                    obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                                }
+                                    break;
+                                case Direction.West:
+                                {
+                                    //facing west, attack north
+                                    possibleTargets.AddRange(
+                                        Map.EntityTree.GetObjects(new Rectangle(this.X, this.Y, 1, 1 + intent.Radius))
+                                            .Where(
+                                                obj =>
+                                                    obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                                }
+                                    break;
+                            }
+                        }
+                            break;
+                        case IntentDirection.Nearby:
+                        {
+                            //attack radius
+                            possibleTargets.AddRange(
+                                Map.EntityTree.GetObjects(new Rectangle(this.X - intent.Radius, this.Y,
+                                        (this.X + intent.Radius) - (this.X - intent.Radius),
+                                        (this.Y + intent.Radius) - (this.Y - intent.Radius)))
+                                    .Where(obj => obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                            possibleTargets.AddRange(
+                                Map.EntityTree.GetObjects(new Rectangle(this.X, this.Y - intent.Radius,
+                                        (this.X + intent.Radius) - (this.X - intent.Radius),
+                                        (this.Y + intent.Radius) - (this.Y - intent.Radius)))
+                                    .Where(obj => obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                        }
+                            break;

```

Comment:
I agree, however that is only moving one piece of a nested call to later in the code. This case statement is large, and nested, due to the minor variations that need to be accounted for based on IntentDirection, followed by User facing.  

I can pull out the `possibleTargets.AddRange(` but that will then still leave me needing to create, and return a new variable for `Map.EntityTree.GetObjects`. 

So while I agree with the want to change this statement, I disagree with the need for it.